app/views/unattended/provisioning_templates/snippet/puppet_setup.erb

```diff
@@ -152,7 +152,24 @@ export FACTER_is_installer=true
 # passing a non-existent tag like "no_such_tag" to the puppet agent only initializes the node
 <%= bin_path %>/puppet agent --config <%= etc_path %>/puppet.conf --onetime <%= host_param_true?('run-puppet-in-installer') || @full_puppet_run ? '' : '--tags no_such_tag' %> <%= @host.puppetmaster.blank? ? '' : "--server #{@host.puppetmaster}" %> --no-daemonize
 <% if os_family == 'Suse' || (os_name == 'Debian' && os_major > 8) || (os_name == 'Ubuntu' && os_major >= 15) -%>
-<%= bin_path %>/puppet resource service puppet enable=true
+<% if os_family == 'Suse' && os_major == 11 -%>
+# workaround for SLES 11.4 where systemd is not installed per default
+START_PUPPET_SCRIPT='/etc/init.d/start_puppet'
+echo "#!/bin/sh
+### BEGIN INIT INFO
+# Provides: puppet_start
+# Required-Start: $all
+# Required-Stop:
+# Default-Start:  2 3 5
+# Default-Stop:
+# Description: Start puppet service
+### END INIT INFO
+puppet resource service puppet enable=true" >> "${START_PUPPET_SCRIPT}"
```

Comment:
Do you really want this enable line inside the init script? It feels weird to do that every single time.

Reply:
Thank you for your feedback. I came up with a more elegant solution.